[CI] CoordinationStateTests testSafety failure

Failed on 7.6 here: https://gradle-enterprise.elastic.co/s/xqj6qfl7477mk

This reproduces on 7.6, and also tried on 7.x and master where it also failed for me locally:
```
./gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.CoordinationStateTests.testSafety" \
  -Dtests.seed=C09AD7BDD00AAC7B \
  -Dtests.security.manager=true \
  -Dtests.locale=ar \
  -Dtests.timezone=America/Mendoza
```

```
java.lang.AssertionError: Multiple masters [{node_0}{0NdpewAAQACzD8aQ_____w}{0.0.0.0}{0.0.0.0:108}{dim}, {node_0}{BpeDEGvzSDacDK6h3VlWeA}{0.0.0.0}{0.0.0.0:108}{dim}] for term 1
Expected: a collection with size <1>
     but: collection size was <2>
	at __randomizedtesting.SeedInfo.seed([C09AD7BDD00AAC7B:F6B17BEDE9B8F0AB]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.elasticsearch.cluster.coordination.CoordinationStateTestCluster.lambda$invariant$10(CoordinationStateTestCluster.java:262)
	at java.util.HashMap.forEach(HashMap.java:1289)
	at org.elasticsearch.cluster.coordination.CoordinationStateTestCluster.invariant(CoordinationStateTestCluster.java:260)
	at org.elasticsearch.cluster.coordination.CoordinationStateTestCluster.runRandomly(CoordinationStateTestCluster.java:253)
	at org.elasticsearch.cluster.coordination.CoordinationStateTests.testSafety(CoordinationStateTests.java:777)
```

--- a/test/framework/src/main/java/org/elasticsearch/cluster/coordination/CoordinationStateTestCluster.java
+++ b/test/framework/src/main/java/org/elasticsearch/cluster/coordination/CoordinationStateTestCluster.java
@@ -106,10 +106,14 @@ public class CoordinationStateTestCluster {
 
         void reboot() {
             if (localNode.isMasterNode() == false && rarely()) {
-                // master-ineligible nodes can't be trusted to persist the cluster state properly
-                persistedState = new InMemoryPersistedState(0L,
-                    clusterState(0L, 0L, localNode, CoordinationMetadata.VotingConfiguration.EMPTY_CONFIG,
-                        CoordinationMetadata.VotingConfiguration.EMPTY_CONFIG, 0L));
+                // master-ineligible nodes can't be trusted to persist the cluster state properly, but will not lose the fact that they
+                // were bootstrapped
+                final CoordinationMetadata.VotingConfiguration votingConfiguration
+                    = persistedState.getLastAcceptedState().getLastAcceptedConfiguration().isEmpty()
+                        ? CoordinationMetadata.VotingConfiguration.EMPTY_CONFIG
+                        : CoordinationMetadata.VotingConfiguration.MUST_JOIN_ELECTED_MASTER;
+                persistedState
+                    = new InMemoryPersistedState(0L, clusterState(0L, 0L, localNode, votingConfiguration, votingConfiguration, 0L));
             }
 
             final Set<DiscoveryNodeRole> roles = new HashSet<>(localNode.getRoles());

